src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1060,67 +1060,66 @@ QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        return convertCharsetInternal(QLatin1String("windows-1253"), value);
     case Turkish:
-        return convertCharsetInternal(QStringLiteral("windows-1254"), value);
+        return convertCharsetInternal(QLatin1String("windows-1254"), value);
     case Hebrew:
-        return convertCharsetInternal(QStringLiteral("windows-1255"), value);
+        return convertCharsetInternal(QLatin1String("windows-1255"), value);
     case Arabic:
-        return convertCharsetInternal(QStringLiteral("windows-1256"), value);
+        return convertCharsetInternal(QLatin1String("windows-1256"), value);
     case Baltic:
-        return convertCharsetInternal(QStringLiteral("windows-1257"), value);
+        return convertCharsetInternal(QLatin1String("windows-1257"), value);
     case Vietnamese:
-        return convertCharsetInternal(QStringLiteral("windows-1258"), value);
+        return convertCharsetInternal(QLatin1String("windows-1258"), value);
     case Latin9:
-        return convertCharsetInternal(QStringLiteral("ISO-8859-15"), value);
+        return convertCharsetInternal(QLatin1String("ISO-8859-15"), value);
     case Shift_JIS:
-        return convertCharsetInternal(QStringLiteral("Shift_JIS"), value);
+        return convertCharsetInternal(QLatin1String("Shift_JIS"), value);
     case EUC_JP:
-        return convertCharsetInternal(QStringLiteral("EUC-JP"), value);
+        return convertCharsetInternal(QLatin1String("EUC-JP"), value);
     case EUC_KR:
-        return convertCharsetInternal(QStringLiteral("EUC-KR"), value);
+        return convertCharsetInternal(QLatin1String("EUC-KR"), value);
     case Big5_HKSCS:
-        return convertCharsetInternal(QStringLiteral("Big5-HKSCS"), value);
+        return convertCharsetInternal(QLatin1String("Big5-HKSCS"), value);
     case KOI8_U:
-        return convertCharsetInternal(QStringLiteral("KOI8-U"), value);
+        return convertCharsetInternal(QLatin1String("KOI8-U"), value);
     case UCS2:
-        return convertCharsetInternal(QStringLiteral("ISO-10646-UCS-2"), value);
+        return convertCharsetInternal(QLatin1String("ISO-10646-UCS-2"), value);
     case SCSU:
-        return convertCharsetInternal(QStringLiteral("SCSU"), value);
+        return convertCharsetInternal(QLatin1String("SCSU"), value);
     case BOCU_1:
-        return convertCharsetInternal(QStringLiteral("BOCU-1"), value);
+        return convertCharsetInternal(QLatin1String("BOCU-1"), value);
     case CESU_8:
-        return convertCharsetInternal(QStringLiteral("CESU-8"), value);
+        return convertCharsetInternal(QLatin1String("CESU-8"), value);
     }
     m_pScriptEngineLegacy->logOrThrowError(QStringLiteral("Unknown charset specified"));
     return QByteArray();
 }
 
 QByteArray ControllerScriptInterfaceLegacy::convertCharsetInternal(
-        const QString& targetCharset, const QString& value) {
+        const QLatin1String& targetCharset, const QString& value) {
```

Comment:
This is supposed to be called from Javascript where `String` is implemented directly in terms of `QString` (which I'm quite certain about). So you're just forcing the Qt runtime to always do the conversion. The conversion is also already avoided in the >=Qt6.8 path. 
Moreover, there is no reason to take a reference type like `QLatin1String` by const ref. So even if we were to introduce it here, we should take it by value.

Reply:
I propose to drop this commit entirely.

Reply:
This is the "private" function convertCharsetInternal(), so we can benefit form no conversion in the < Qt6.8 case as well.

Reply:
Ah I see. You're correct.